### PR TITLE
support run e2e in rosa env

### PR DESCRIPTION
**Pull request by skeeey**

### CODE REVIEW

File: test/e2e/pkg/status_resync_test.go [hunk 1]
```diff
@@ -171,8 +171,10 @@ var _ = Describe("Status Resync After Restart", Ordered, Label("e2e-tests-status
 				return fmt.Errorf("nginx deployment still exists")
 			}, 2*time.Minute, 2*time.Second).ShouldNot(HaveOccurred())
 
-			err = agentTestOpts.kubeClientSet.CoreV1().ServiceAccounts("default").Delete(ctx, name, metav1.DeleteOptions{})
-			Expect(err).ShouldNot(HaveOccurred())
+			// In a ROSA cluster the sa in default namespace is protected, the sa cannot be deleted directly
+			// see https://access.redhat.com/solutions/7055600
+			// err = agentTestOpts.kubeClientSet.CoreV1().ServiceAccounts("default").Delete(ctx, name, metav1.DeleteOptions{})
```
Comment (clyang82) [inline test/e2e/pkg/status_resync_test.go:176]: Do we create the `default` serviceaccount? I think it should be there by default.
@morvencao comments?

Comment (morvencao) [reply test/e2e/pkg/status_resync_test.go:176]: we assume `default` sa is there, it's not created by us.
the code change looks good to me, we can remove this.

Comment (clyang82) [reply test/e2e/pkg/status_resync_test.go:176]: Thanks @morvencao 
@skeeey Let us remove it directly

Comment (skeeey) [reply test/e2e/pkg/status_resync_test.go]: removed

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**